python/setup.py

```diff
@@ -100,6 +100,7 @@ def run(self):
               sources=CYTHON_FILES,
               include_dirs=[conda_include_dir,
                             '../cpp/include',
+                            '../cpp/src',
```

Comment:
Better encapsulate the public C++ API in a public header and move it to `../cpp/include`.
Files under `../cpp/src` are internals and don't get deployed.